### PR TITLE
Setup SignInWithSlack button

### PR DESCRIPTION
**Pull request by nobioma1**
**What does PR do**
- Create SignInWithSlack button component
- Add transition to AddToSlackbutton
- Create Protected Route Component
- Remove querystring and install query-string
- Create Test Suites

### CODE REVIEW

File: src/components/SlackButton/index.js [hunk 1]
```diff
@@ -11,9 +11,10 @@ import {
 import slackLogo from '../../assets/slack-logo.png';
 import arrow from '../../assets/arrow.png';
 
+const clientId = process.env.REACT_APP_CLIENT_ID;
+const redirectURI = `${document.location.origin}/slack`;
```
Comment (ropeksdev) [inline src/components/SlackButton/index.js]: Nice drying Noble! Btw, why not extract out scope as well?

Comment (ropeks) [inline src/components/SlackButton/index.js]: Nice drying Noble! Btw, why not extract out scope as well?

File: src/utils/slackOAuth.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+    }
+  );
+  if ('error' in slackRes) {
+    throw new Error(slackRes.error);
+  }
+  return slackRes;
+};
```
Comment (ropeks) [inline src/utils/slackOAuth.js]: 👌

File: src/components/SlackButton/index.js (hunk 1 above)
Comment (nobioma1) [reply src/components/SlackButton/index.js]: @ropeks the scopes for each of the button is different.
-  Sign in with Slack Button: 
`const scope = 'identity.basic,identity.email,identity.team,identity.avatar';`
- Add to Slack button: 
`const scope = 'incoming-webhook,commands';`

Comment (ropeks) [reply src/components/SlackButton/index.js]: @nobioma1 thanks for clearing that up!

**Pull request merged**